### PR TITLE
Fix/limit fridge configuration to 5 stages

### PR DESCRIPTION
**Pull request by JohnCLee7**
- Implemented feature flags
- Implemented feature flag to indicate app should only accept fridge configurations with 5 stages, removing stage addition and deletion controls, and updating import to refuse configurations that do not have exactly 5 stages.

### CODE REVIEW

File: components/config/featureFlagContext.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+"use client"
+
+import { createContext, FC, PropsWithChildren, useContext, useEffect, useState } from "react";
+
+export type FeatureFlags = {
+    staticStageCount: boolean
+}
+
+const featureFlagsContext = createContext<FeatureFlags>({ staticStageCount: true });
```
Comment (ZyrnDev) [inline components/config/featureFlagContext.tsx]: make the default fridge a single variable

File: components/config/featureFlagContext.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+export const useFeatureFlags = () => useContext(featureFlagsContext);
+
+export const FeatureFlagProvider: FC<PropsWithChildren> = ({ children }) => {
+
+    const [featureFlags, setFeatureFlags] = useState<FeatureFlags>({
+        staticStageCount: true
+    });
```
Comment (ZyrnDev) [inline components/config/featureFlagContext.tsx]: make the default fridge a single variable

File: components/config/featureFlagContext.tsx (hunk 1 above)
Comment (ZyrnDev) [inline components/config/featureFlagContext.tsx]: Capitalize the first Letter of FeatureFlagsContext as is is standard

**Review by ZyrnDev — CHANGES_REQUESTED** [review]

File: app/layout.tsx [hunk 3]
```diff
@@ -19,8 +20,10 @@ export default function RootLayout({
       <body>
         <PythonRuntimeProvider>
           <FridgeProvider>
+            <FeatureFlagProvider>
               <NavBar />
               {children}
+            </FeatureFlagProvider>
```
Comment (ZyrnDev) [inline app/layout.tsx]: please make `FeatureFlagsProvider` an ancestor of `FridgeProvider` if you are going to use the feature flags in the `FridgeProvider`

**Review by ZyrnDev — CHANGES_REQUESTED** [review]

File: components/config/featureFlagContext.tsx (hunk 1 above)
Comment (JohnCLee7) [reply components/config/featureFlagContext.tsx]: Added

File: components/config/featureFlagContext.tsx (hunk 2 above)
Comment (JohnCLee7) [reply components/config/featureFlagContext.tsx]: Added

File: components/config/featureFlagContext.tsx (hunk 1 above)
Comment (JohnCLee7) [reply components/config/featureFlagContext.tsx]: Done

File: app/layout.tsx (hunk 3 above)
Comment (JohnCLee7) [reply app/layout.tsx]: Done

**Review by ZyrnDev — APPROVED** [review]

**Pull request merged**